server.js

```diff
@@ -21,7 +22,47 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Hello Book Entusiats!");
+});
+
+app.get("/books", (req, res) => {
+  res.status(200).json({
+    data: booksData,
+    success: true,
+  });
+})
+
+app.get("/books/title/:title", (req, res) => {
+  const { title } = req.params;
+
+  const bookByName = booksData.find(
+    (book) => book.title.toLowerCase() === title.toLowerCase()
+  );
+
+  if (!bookByName) {
+    res.status(404).json({
+      data: "Not found",
+      success: false,
+    });
+  } else {
+    res.status(200).json({
+      data: bookByName,
+      success: true,
+    });
+  }
+});
+
+app.get("/books/author/:author", (req, res) => {
+  const { author } = req.params;
+
+  const booksByAuthor = booksData.filter(
+    (book) => book.authors.toLowerCase() === author.toLowerCase()
```

Comment:
To make the search action return some results without the need to type entire name/title we could use the include method. It would look something like this: 
`let booksByAuthor = booksData.filter(
    (book) => book.authors.toLowerCase()
    .includes(authors.toLocaleLowerCase()));`
   This could help the user type just a part of the name and it would eventually give the user more than one result back.